shell/server/proxy.js

```diff
@@ -304,16 +313,10 @@ function openSessionInternal(grainId, user, title, apiToken) {
 
   var isOwner = grainInfo.owner === userId;
 
-  var sessionId = Random.id();
-  var proxy = new Proxy(grainId, grainInfo.owner, sessionId, null, isOwner, user, null, false);
-  proxy.apiToken = apiToken;
-  proxies[sessionId] = proxy;
```

Comment:
Looks like you are no longer setting `proxies[sessionId]` anywhere. Either you should eradicate that map completely (it's currently used by `getGrainSize()`) or you should update it when you update `proxiesByHostId`.